bench/src/main.rs

```diff
@@ -39,17 +40,33 @@ async fn main() {
 
     let mut avg_metric = AvgMetric::default();
 
-    for run_num in 0..runs {
-        let metric = bench(rpc_client.clone(), tx_count).await;
-        info!("Run {run_num}: Sent and Confirmed {tx_count} tx(s) in {metric:?} with",);
-        // update avg metric
-        avg_metric += &metric;
-        // write metric to file
-        csv_writer.serialize(metric).unwrap();
+    let mut tasks = vec![];
+
+    for _ in 0..runs {
+        let rpc_client = rpc_client.clone();
+        tasks.push(tokio::spawn(bench(rpc_client.clone(), tx_count)));
         // wait for an interval
         run_interval_ms.tick().await;
     }
 
+    let join_res = join_all(tasks).await;
+
+    let mut run_num = 1;
+    for res in join_res {
+        match res {
+            Ok(metric) => {
+                info!("Run {run_num}: Sent and Confirmed {tx_count} tx(s) in {metric:?} with",);
+                // update avg metric
+                avg_metric += &metric;
```

Comment:
This is not an accurate way top calculate avg.
cause join_all waits for all futures

Reply:
Ok i will update it later.